arbos/arbosState/arbosstate.go

```diff
@@ -379,11 +386,17 @@ func (state *ArbosState) UpgradeArbosVersion(
 			// these versions are left to Orbit chains for custom upgrades.
 
 		case params.ArbosVersion_50:
-			params, err := state.Programs().Params()
+			p, err := state.Programs().Params()
 			ensure(err)
-			ensure(params.UpgradeToArbosVersion(nextArbosVersion))
-			ensure(params.Save())
-
+			ensure(p.UpgradeToArbosVersion(nextArbosVersion))
+			ensure(p.Save())
+			chainId, err := state.ChainId()
+			ensure(err)
+			if chainId.Cmp(chaininfo.ArbitrumOneChainConfig().ChainID) == 0 || chainId.Cmp(chaininfo.ArbitrumNovaChainConfig().ChainID) == 0 {
+				ensure(state.l1PricingState.SetCalldataPrice(big.NewInt(int64(params.TxCostFloorPerToken))))
+			} else {
+				ensure(state.l1PricingState.SetCalldataPrice(big.NewInt(int64(params.TxDataNonZeroGasEIP2028))))
+			}
```

Comment:
you also need to update state.l1PricingState.arbosVersion..
best way is to do it in current line ~416 - right after:
```
state.programs.ArbosVersion = nextArbosVersion
```